### DISCUSSION TITLE
How to add a qiskit backend with a Noise Model?

### DISCUSSION THREAD

**Discussion by inajetovic** [entropicalabs/openqaoa#316] [Q&A]
Hi all, 
I'm new to OpenQAOA but I'm really interested in its features! I'm currently working on a Noise Model with qiskit. Do you have some recommendations about how to easily add a local qiskit backend featured by a custom noise model (in particular a AerSimulator noisy backend)? Should I create a new class or there's a more straightforward way to do it?

**Comment by vishal-ph** [comment]
Hi @inajetovic, if I understand your question correctly, you have a custom noise model that you wish to use with the qiskit AerSimulator in openqaoa. If that is the case, you need not add a new backend specifically. You can simply pass your custom noise model as an argument to the existing AerSimulator in the OpenQAOA backend as follows:

```{python}
q_noisy_backend = QAOA()
qiskit_noisy_simulator = create_device(location='local', name='qiskit.qasm_simulator', noise_model = <YOUR NOISE MODEL>)
q_noisy_backend.set_device(qiskit_noisy_simulator)
```
Do keep in mind, that your custom noise model should be of type `NoiseModel` as that is what the qiskit simulator supports.
